solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpClientUtil.java

```diff
@@ -447,14 +448,39 @@ public void process(final HttpResponse response, final HttpContext context)
     }
   }
   
-  private static class GzipDecompressingEntity extends HttpEntityWrapper {
+  protected static class GzipDecompressingEntity extends HttpEntityWrapper {
+    private boolean gzipInputStreamCreated = false;
+    private InputStream gzipInputStream = null;
+
     public GzipDecompressingEntity(final HttpEntity entity) {
       super(entity);
     }
-    
+
+    /**
+     * Return a InputStream of uncompressed data.
+     * If there is an issue with the compression of the data, a null InputStream will be returned,
+     * and the underlying compressed InputStream will be closed.
+     *
+     * The same input stream will be returned if the underlying entity is not repeatable.
+     * If the underlying entity is repeatable, then a new input stream will be created.
```

Comment:
do we have a test that covers this? what is the use case for repeatable entities?

Reply:
I'm not exactly sure. I just included the option so that I don't have to confirm that we can expect a repeatable entity every time. I can add a better test for it, instead of just testing the failures

Reply:
Tests for failing & successful, repeatable & non-repeatable enumerated.